components/brave_wallet_ui/common/async/lib.ts

```diff
@@ -931,7 +932,7 @@ export async function sendEthTransaction (payload: SendEthTransactionParams) {
 
     // Check if network and keyring support EIP-1559.
     default:
-      isEIP1559 = hasEIP1559Support(payload.fromAccount.accountType, payload.network)
+      isEIP1559 = hasEIP1559Support(getAccountType(payload.fromAccount), payload.network)
```

Comment:
There is only a few places where we need Primary/Secondary/Ledger/Trezor account type, so calculate that on demand